### PR TITLE
Pre-4.7.0 check.

### PR DESCRIPTION
**Pull request by asmorkalov**

### CODE REVIEW

**Comment by asmorkalov** [comment]
@alalek there is linker issue on Windows:
```
     Creating library D:/a/opencv-python/opencv-python/_skbuild/win-amd64-3.7/cmake-build/lib/Release/opencv_python3.lib and object D:/a/opencv-python/opencv-python/_skbuild/win-amd64-3.7/cmake-build/lib/Release/opencv_python3.exp
cv2.obj : error LNK2019: unresolved external symbol "public: virtual class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl cv::xfeatures2d::SURF::getDefaultName(void)const " (?getDefaultName@SURF@xfeatures2d@cv@@UEBA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ) referenced in function "[thunk]:public: virtual class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl cv::xfeatures2d::SURF::getDefaultName`vtordisp{4294967292,0}' (void)const " (?getDefaultName@SURF@xfeatures2d@cv@@$4PPPPPPPM@A@EBA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ) [D:\a\opencv-python\opencv-python\_skbuild\win-amd64-3.7\cmake-build\modules\python3\opencv_python3.vcxproj]
D:\a\opencv-python\opencv-python\_skbuild\win-amd64-3.7\cmake-build\lib\python3\Release\cv2.pyd : fatal error LNK1120: 1 unresolved externals [D:\a\opencv-python\opencv-python\_skbuild\win-amd64-3.7\cmake-build\modules\python3\opencv_python3.vcxproj]
```
Looking on it right now.

**Comment by asmorkalov** [comment]
Fix on WIndows: https://github.com/opencv/opencv_contrib/pull/3403

File: scripts/install.sh [hunk 1]
```diff
@@ -15,5 +15,8 @@ echo $ENABLE_CONTRIB > contrib.enabled
 echo $ENABLE_HEADLESS > headless.enabled
 echo $ENABLE_ROLLING > rolling.enabled
 set -x
+if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+  pip install --upgrade pip
+fi
```
Comment (asenyaev) [inline scripts/install.sh]: Let's move it to `travis_config.sh` somewhere [here](https://github.com/asmorkalov/opencv-python/blob/as/pre-4.7.0/travis_config.sh#L147). In case the issue with pip only with Python 3.7.
```
created virtual environment CPython3.7.13.final.0-64 in 452ms
  creator CPython3Posix(dest=/venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, wheel=bundle, pip=bundle, setuptools=bundle, via=copy, app_data_dir=/root/.local/share/virtualenv)
    added seed packages: pip==19.1.1, setuptools==43.0.0, wheel==0.33.6
```

Comment (asenyaev) [reply scripts/install.sh]: It works well in the current status of `4.x` branch:
```
created virtual environment CPython3.7.15.final.0-64 in 871ms
  creator CPython3Posix(dest=/venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/root/.local/share/virtualenv)
    added seed packages: pip==22.3.1, setuptools==65.5.1, wheel==0.38.4
```

The main reason - the docker image for tests is different. Idk why it has changed in this PR, because you updates multibuild submodule and it shouldn't happen.

The wrong image (this had been using for some time and was dropped half a year ago) - `matthewbrett/trusty:64`

The proper image - `multibuild/focal_x86_64`

This rule should work here, but it doesn't ([link](https://github.com/multi-build/multibuild/blob/devel/travis_linux_steps.sh#L131))
```
local docker_image="multibuild/focal_{PLAT}"
```

**TL;DR**
While I was writing this comment, I found out that commit for multibuild submodule is too old (2020 year). Move back to the previous version which was stable or update to the latest one (it also can solve Python 3.11 issues).

File: patch_auditwheel_whitelist.py [hunk 2]
```diff
@@ -15,6 +15,7 @@ def add_zlib_versions():
                 updated_line = line[:-2] + ', "1.2.9", "1.2.12"]'
             else:
                 updated_line = line[:-2] + '"1.2.9", "1.2.12"]'
```
Comment (asenyaev) [inline patch_auditwheel_whitelist.py]: Try to add `1.2.13` version for Zlib also, because it has been changed in a docker to `1.2.13`. It may solve an issue with Zlib in tests (but better to start from a new docker image for tests).

I don't have an opportunity to test it on Linux and with docker now, but as I remember a whitelist was different from time to time and maybe some versions should be added also (I hope not).

**Comment by asenyaev** [comment]
At the moment also the old multibuild commit (2021 year). Let's use the current one in `4.x` branch - [bce1637e202cb52b0e7ea42baa1cccc894b82806](https://github.com/multi-build/multibuild/tree/bce1637e202cb52b0e7ea42baa1cccc894b82806)

The PR where it was added - https://github.com/opencv/opencv-python/pull/684

The change in a submodule with a test docker image was committed in March this year ([change](https://github.com/multi-build/multibuild/commit/86c0bbca2c18df12b684f7ca441c5255506a7268#diff-bd67d03d93262cdff22c2340735298b36446194aaea879a084976d51c5b886aaL120-R123)).

File: patch_auditwheel_whitelist.py [hunk 3]
```diff
@@ -12,9 +12,10 @@ def add_zlib_versions():
     for line in manylinux_policy_lines:
         if "ZLIB" in line:
             if len(line) > 22:
-                updated_line = line[:-2] + ', "1.2.9", "1.2.12"]'
+                updated_line = line[:-2] + ', "1.2.9", "1.2.12", "1.2.13"]'
             else:
-                updated_line = line[:-2] + '"1.2.9", "1.2.12"]'
+                updated_line = line[:-2] + '"1.2.9", "1.2.12", "1.2.12"]'
```
Comment (asenyaev) [inline patch_auditwheel_whitelist.py]: "1.2.9", "1.2.12", "1.2.1**3**"

**Review by asenyaev — APPROVED** [review]
